src/Informatik/EinfuehrungProgrammierung.tex

```diff
@@ -51,20 +51,95 @@
         \paragraph{Funktionsaufruf}
             Bei strikter Auswertung werden erst alle Argumente eines Funktionsaufrufs berechnet. Dieses Verfahren kommt bei Racket zum Einsatz. Diesem Verfahren steht die lazy Auswertung gegenüber, welche zum Beispiel in Haskel verwendet wird, bei dem auch die Argumente erst berechnet werden, wenn sie benötigt werden.   
             
-    \section{Aufschreibregeln für Funktionen}
+    \section{Aufschreibregeln}
         \paragraph{Regel 1}
             Verwende aussagekräftige Namen für Funktionen und deren Parameter.
         
         \paragraph{Regel 2}
             Beschreibe den Zweck der Funktion in einem Satz mithilfe eines Kommentars in der Zeile vor der Funktionsdefinition. Der Satz sollte das Resultat der Funktion in Abhängigkeit von den Parametern beschreiben.
             
         \paragraph{Regel 3}
-            Füge unter der Funktion Tests hinzu, die die Richtigkeit der Funktion auch in Zukunft sicherstellen. Dazu müssen Test für alle Randbereiche vorhanden sein.
+            Zwischen Funktionesbeschreibung und Funktionskopf werden Tests hinzugefügt, die die Richtigkeit der Funktion auch in Zukunft sicherstellen. Dazu müssen Test für alle Randbereiche vorhanden sein.
             
         \paragraph{Regel 4}
             Definiere für jeden Zusammenhang zwischen Größen, die sich aus der Problemstellung ergeben, eine Funktion.
             
         \paragraph{Regel 5}
             Ersetze jede Konstante, deren Bedeutung sich nicht aus dem Kontext ergibt, durch einen sprechenden Variablennamen.
+            
+        \paragraph{Regel 6}
+            Anstatt den Funktionsrumpf direkt zu schreiben, schreibt man ein Skelett, das man anschließend mit Inhalt füllt. Das Skelett kann zum Beispiel von einer cond Verzweigung oder einem Konstruktor sein.
+            
+        \paragraph{Regel 7}
+            Vor Strukturdefinitionen wird ähnlich wie bei Funktionen eine Datendefinition der Form
+            
+            \begin{lstlisting}[language=Lisp]
+                ;; Ein s ist ein Wert
+                ;; (make-s s-1 ... s-n)
+                ;; wobei s-1 ...
+            \end{lstlisting}
+            
+    \section{Datenabstraktion}
+        \subsection{Abstraktionsbarrieren}
+            Beispiel 
+            
+            \begin{itemize}
+                \item \textbf{Punkt in der Sicht der Anwendungsprogramme}
+                \item ... add-point equal-point? ...
+                \item \textbf{Punkte bestehen aus x- und y-Koordinaten}
+                \item ... make-point point-x point-y
+                \item \textbf{Punkt implementiert als pairs}
+                \item ... cons car cdr ...
+                \item \textbf{Interne Implementation von pairs}
+            \end{itemize}
+        
+        \subsection{define-struct}
```

Comment:
`\begin{quote}` in this subsection should be replaced w/ `\begin{lstlisting}[language=Lisp]`